### ISSUE TITLE
Export float conversion routines (or convert exported data into go types)

### ISSUE THREAD

**Issue by nikolaydubina** [redis/go-redis#3206]
very important value `__score` in `FT.SEARCH` is returned as raw string not converted. users need manually convert it to float

now, this is internal redis type.

and redis go pkg does not export functions to convert it. which is bad and error prone and leads to wrong re-implementations.

## Expected Behavior

Either:

- A) provide utilities to convert internal string float representations
- B) export standard float notation (and rest of types)
- C) export normal native go float in exposed functions

## Current Behavior

<img width="1424" alt="image" src="https://github.com/user-attachments/assets/631a1dd4-b025-41bc-a39e-8070bb8b73fe">

<img width="373" alt="image" src="https://github.com/user-attachments/assets/21963e73-e43c-43c8-9323-6bad3455f440">

## Possible Solution

export this https://github.com/redis/go-redis/blob/master/internal/proto/reader.go#L192

## Steps to Reproduce

1. use `FT.SEARCH` `__score` field in Go
2. try to convert it to float
3. search for any methods that redis provides to convert it (there is none)

## Context (Environment)

`FT.SEARCH` is very important. `__score` is very important.

and this is not even proper float... what....

`INFO`

```
# Server
redis_version:7.2.6
redis_git_sha1:00000000
redis_git_dirty:0
redis_build_id:afdd7ad38aac5fcf
redis_mode:standalone
os:Linux 6.1.112+ x86_64
arch_bits:64
monotonic_clock:POSIX clock_gettime
multiplexing_api:epoll
atomicvar_api:c11-builtin
gcc_version:11.4.0
process_id:8

```

```
go version go1.23.0 darwin/arm64
```

go.mod
```
github.com/redis/go-redis/v9 v9.7.0
```

**Comment by SoulPancake** [comment]
Can I take this up @ndyakov ?

**Comment by ndyakov** [comment]
Sure, if you need any guidance let me know.

**Comment by SoulPancake** [comment]
@ndyakov Should we export readFloat util? @ndyakov

**Comment by ndyakov** [comment]
@SoulPancake yes, I assume we can create a second exported package utils and have a function that will execute the internal function. But just that, no need to export the whole utils package that is currently placed in `internal`.

**Comment by SoulPancake** [comment]
@ndyakov 
Do you mean we create a new utils package which will just call the utils in internal?

However that would also require us to export the function so that we can call the internal function
Also when you say exported package, does it have to be with it's own go-module so it can be exported separately?

**Comment by SoulPancake** [comment]
@ndyakov ^?
